iml-wire-types/wire-types.rs

```diff
@@ -1235,3 +1235,49 @@ pub mod db {
         }
     }
 }
+
+// Types used for component checks
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+pub struct ElementState {
+    name: String,
+    configurable: bool,
+}
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+pub enum ServiceState {
+    Unconfigured,
+    Configured,
+    // Following assume Configured
+    Enabled,
+    Started,
+    Setup, // Started + Enabled
+}
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+pub enum ConfigState {
+    Unknown,
+    Default, // components is in default configuration
+    IML,     // matches what IML would do
+    Other,
+}
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+pub struct ComponentState<T> {
+    service: ServiceState,
+    config: ConfigState,
+    elements: Vec<ElementState>,
+    info: String,
+    state: T,
+}
+
+impl<T> ComponentState<T> {
+    pub fn new(val: T) -> Self {
```

Comment:
Since most of these assume a default, should we just implement the `Default` trait instead?

Reply:
Yes, I will update